### ISSUE TITLE
TS1117 doesn't catch duplicate Symbols

### ISSUE THREAD

**Issue by JeremyLoy** [microsoft/TypeScript#55163]
# Bug Report

> error TS1117: An object literal cannot have multiple properties with the same name in strict mode.

I believe that TS1117 should catch duplicate Symbols in strict mode, or alternatively a new rule should be made to catch duplicate Symbols in an object literal.

### 🔎 Search Terms

symbol
ts1117
### 🕗 Version & Regression Information

- This is the behavior in every version I tried, and I reviewed the FAQ for entries about __symbols__

### ⏯ Playground Link

https://www.typescriptlang.org/play?#code/FASwdgLgpgTgZgQwMZQAQHkAOAVAnpqAZ1QG9hVUYoEATAezABtdUoBHAVwUYC5UOwITmkK4AtgCM6jYAF9gwJA0IQMmPljwFiAXlLkK7Lr1QBlcVMYA6OHRgAKAORHujgJRzUCYpvxEFSmAqqBAAFiCEAMJ2VEgQzABiCCCMuvoUhpzcfI5wIDAqjgA0BpnGOYRQgTTFpagA2s5ZjI4AujmM3hCOcgHKqmERACJ0RJBJKQBCHBCmoXQcjDSoemQZDVhWLoztqLn5hSXr9Zvbu46dhb0KAPQ3qFJhqJgw4KoXXY5eYMsMzKihBAANzQDDQLzoBBgEFwwDuqCUMFiqgkUEBQJAdiKqHAiORIT8ALexFsMBCoREFmkrAAHggxJhGFA+kFpFArIw6ABzeyDKIxKrxXATVJbZoeRTKNkc7m88KEEZjCAi6azeaLGgnTBi4ytNxAA

### 💻 Code

```ts

interface OpTypes {
  readonly equal: unique symbol
}

const Op: OpTypes = {
   equal: Symbol.for('equal')
} as OpTypes

const thisCorrectlyFails = {
    equal: 'first',
    equal: 'second',
    ['equal']: 'last'
}

const thisDoesntFailButShould = {
    [Op.equal]: 'first',
    [Op.equal]: 'last'
}


// both print 'last' and only have one property
// correct behavior, incorrect type hints for the symbol example

console.log(thisCorrectlyFails.equal)

console.log(thisDoesntFailButShould[Op.equal])
```

### 🙁 Actual behavior

TS1117 didn't catch the error in the Symbol example

### 🙂 Expected behavior

For the duplicate Symbol to trigger TS1117

## Context

We use Sequelize as an ORM, and the basic operations (`AND`, `OR` etc) are all `symbols`. We recently fixed a bug in our system where we were overriding an operator that TS didn't catch

https://sequelize.org/docs/v6/core-concepts/model-querying-basics/#operators

**Comment by Zzzen** [comment]
related to #55084 29718

**Comment by JeremyLoy** [comment]
Thanks @Zzzen !
